### ISSUE TITLE
gradle-wrapper.jar outdated

### ISSUE THREAD

**Issue by IzzySoft** [jabstone/JABtalk#10]
Your `/gradle/wrapper/gradle-wrapper.jar` is 9 years old and no longer matching current gradle versions. While you just updated the `gradle-wrapper.properties`, you probably forgot to upgrade the JAR as well. Could you please do so? Thanks in advance!

If you wonder about the background: JABtalk is [listed at IzzyOnDroid](https://apt.izzysoft.de/packages/com.jabstone.jabtalk.basic). At IzzyOnDroid we support [Reproducible Builds](https://reproducible-builds.org/) (see: [Reproducible Builds, special client support and more at IzzyOnDroid](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients)). JABtalk succeeded as such (congrats!) and thus its listing shows the "green shield" confirming this. Currently, whenever you change the gradle version, the recipe at the verification builder requires a manual update as your JAR is not recognized – which leads to delays of confirmation, or to a release being marked "RB failed". An up-to-date `gradle-wrapper.jar` easily avoids that :wink: So thanks for your support!

**Comment by jabstone** [comment]
I just updated it and pushed the changes. If it still doesn't work, I'll need more info on how to upgrade the file correctly.

**Comment by IzzySoft** [comment]
That was fast, thanks! I've checked but funnily still get

```
+ /opt/gradle-wrapper-verify/gradle-wrapper-verify ./gradle/wrapper/gradle-wrapper.jar
checking ./gradle/wrapper/gradle-wrapper.jar ...
unknown hash
```

even when building from the latest commit (80aeb41726858201071ca511e551cd3a7a64a913 – which replaced the JAR). Where did you take the JAR from?

cc @obfusk (who wrote the `gradle-wrapper-verify` tool)

**Comment by jabstone** [comment]
I regenerated it via `./gradlew wrapper --gradle-version 8.10.1`.  I followed these directions:

`Replace the gradle-wrapper.jar
To ensure your gradle-wrapper.jar file is up-to-date, you can regenerate it using the Gradle Wrapper command.

Here are the steps:

Open a terminal in your project’s root directory.
Run the following command to update the wrapper JAR and other related files:

./gradlew wrapper --gradle-version [VERSION]
Replace [VERSION] with the latest Gradle version (e.g., 8.10.1).`

**Comment by obfusk** [comment]
@IzzySoft the issue seems to be on your end:

```
$ git clone https://github.com/jabstone/JABtalk
$ cd JABtalk
$ git checkout 80aeb41726858201071ca511e551cd3a7a64a913
$ gradle-wrapper-verify
checking gradle/wrapper/gradle-wrapper.jar ...
OK
```

**Comment by IzzySoft** [comment]
@obfusk funny, that's the recipe I've used:

```yaml
  - tag: 5.3.2
    apks:
      - apk_pattern: jabtalk.*\.apk
        apk_url: https://github.com/jabstone/JABtalk/releases/download/5.3.2/jabtalk_v5.3.2.apk
        build:
          - git checkout 80aeb41726858201071ca511e551cd3a7a64a913
          - ./gradlew assembleRelease
          - mv app/build/outputs/apk/release/*unsigned.apk /outputs/unsigned.apk
```

Let me try again… Nope, fails. Looking at the logs I miss output from `git checkout`. Eek: of course gradle is processed **before** the build recipe is started with – so the checkout in the recipe comes too late :man_facepalming: Let me try with `com.jabstone.jabtalk.basic:5.3.2:80aeb41726858201071ca511e551cd3a7a64a913:none`…

:bow:

PEBKAC. Apologies, @jabstone – all seems fine now! This time the error indeed was on my end. Should have thought of that…
